fbpic/openpmd_diag/field_diag.py

```diff
@@ -163,21 +165,69 @@ def write_dataset( self, field_grp, path, quantity ) :
             dset = None
 
         # Write the mode 0 : only the real part is non-zero
-        mode0 = self.get_dataset( quantity, 0 )
+        mode0 = self.get_dataset( quantity, 0)
         if self.rank == 0:
             mode0 = mode0.T
             dset[0,:,:] = mode0[:,:].real
         # Write the higher modes
         # There is a factor 2 here so as to comply with the convention in
         # Lifschitz et al., which is also the convention adopted in Warp Circ
         for m in range(1,self.fld.Nm):
-            mode = self.get_dataset( quantity, m )
+            mode = self.get_dataset( quantity, m)
             if self.rank == 0:
                 mode = mode.T
                 dset[2*m-1,:,:] = 2*mode[:,:].real
                 dset[2*m,:,:] = 2*mode[:,:].imag
 
-    def get_dataset( self, quantity, m ):
+    def write_dataset_envelope( self, field_grp, path, quantity):
+        """
+        Write a given dataset
+
+        Parameters
+        ----------
+        field_grp : an h5py.Group object
+            The group that corresponds to the path indicated in meshesPath
+
+        path : string
+            The relative path where to write the dataset, in field_grp
+
+        quantity : string
+            Describes which envelope is being written.
+            (Either A or dtA)
+        """
+        path_real = "%s_real" %(path)
+        path_imag = "%s_imag" %(path)
+        if field_grp is not None:
+            dset_real = field_grp[path_real]
+            dset_imag = field_grp[path_imag]
+        else:
+            dset_real = None
+            dset_imag = None
+
+        mode0 = self.get_dataset(quantity, 0, envelope = True)
+        if self.rank == 0:
+            mode0 = mode0.T
+            dset_real[0,:,:] = mode0[:,:].real
+            dset_imag[0,:,:] = mode0[:,:].imag
+
+        for m in range(1,self.fld.Nm):
+            modep = self.get_dataset( quantity, m, envelope = True)
+            modem = self.get_dataset( quantity, -m, envelope = True)
+            real_mode = 0.5 * (modep + modem.conjugate())
+            imag_mode = -0.5 * 1j * (modep - modem.conjugate())
+            if self.rank == 0:
+                real_mode = real_mode.T
+                imag_mode = imag_mode.T
+                # There is a factor 2 here so as to comply with the convention
+                # in Lifschitz et al., which is also the convention
+                # adopted in Warp Circ
+                dset_real[2*m-1,:,:] = 2*real_mode[:,:].real
+                dset_real[2*m,:,:] = 2*real_mode[:,:].imag
+                dset_imag[2*m-1,:,:] = 2*imag_mode[:,:].real
+                dset_imag[2*m,:,:] = 2*imag_mode[:,:].imag
+
+
+    def get_dataset( self, quantity, m, envelope = False ):
```

Comment:
Instead of adding the optional envelope argument, could simply do a check on quantity inside this function?
```
if quantity in ['A', 'dtA']:
    data_one_proc = getattr( self.fld.envelope_interp[m], quantity )
else:
    data_one_proc = getattr( self.fld.interp[m], quantity )